src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceConnector.java

```diff
@@ -93,4 +99,35 @@ public void stop() {
 	public String version() {
 		return AppInfoParser.getVersion();
 	}
+	
+	/***
+	 * Get the number of topics from config file
+	 * @return
+	 */
+	private int numberOfTopics() {
+		return configProperties.get(InterfaceConst.TOPIC).split(",").length;
+	}
+	
+	/***
+	 * Generate topics to be consumed considering number of 
+	 * tasks 
+	 * @param maxTasks - max number of tasks
+	 * @return
+	 */
+	private String[] balanceTopicsTasks(int maxTasks) {
+		
+		// Get topics
+		String[] topics = configProperties.get(InterfaceConst.TOPIC).split(",");
+		
+		int topicsLength = numberOfTopics();  
+		// minor memory optimization
+		int size = (topicsLength < maxTasks ? topicsLength : maxTasks );
+		// Set topics in tasks
+		String[] topicsInTask = new String[size];
```

Comment:
Do you think we can use an ArrayList<String> here instead of the String[] ? I get that you manage the array size well but if anything goes wrong we are more likely to see index-out-of-bound exceptions with a fixed size array compared to a dynamic array. This maybe just my style but I try and stay away from fixed arrays like that.